tests/testutils.py

```diff
@@ -19,14 +19,50 @@
 middleware = ComponentDependencyMiddleware(get_response=lambda _: response_stash)
 
 
+class GenIdPatcher:
+    def __init__(self):
+        self._gen_id_count = 10599485
+
+    # Mock the `generate` function used inside `gen_id` so it returns deterministic IDs
+    def start(self):
+        # Random number so that the generated IDs are "hex-looking", e.g. a1bc3d
+        self._gen_id_count = 10599485
+
+        def mock_gen_id(*args, **kwargs):
+            self._gen_id_count += 1
+            return hex(self._gen_id_count)[2:]
+
+        self._gen_id_patch = patch("django_components.util.misc.generate", side_effect=mock_gen_id)
+        self._gen_id_patch.start()
+
+    def stop(self):
+        self._gen_id_patch.stop()
+        self._gen_id_count = 10599485
+
+
+class CsrfTokenPatcher:
```

Comment:
Also the large test also renders a CSRF token, so I've patched that too, so it's replaced with `"predictabletoken"`.